### PR TITLE
[Instrumentation.Runtime] Enable nullable

### PR DESCRIPTION
**Pull request by Kielek**
Towards #894

## Changes

Nullable for Runtime projects.

For significant contributions please make sure you have completed the following items:

* ~~[ ] Appropriate `CHANGELOG.md` updated for non-trivial changes~~
* ~~[ ] Design discussion issue #~~

### CODE REVIEW

**Review by TimothyMothra — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Runtime/MeterProviderBuilderExtensions.cs [hunk 1]
```diff
@@ -33,7 +33,7 @@ public static class MeterProviderBuilderExtensions
     /// <returns>The instance of <see cref="MeterProviderBuilder"/> to chain the calls.</returns>
     public static MeterProviderBuilder AddRuntimeInstrumentation(
         this MeterProviderBuilder builder,
-        Action<RuntimeInstrumentationOptions> configure = null)
+        Action<RuntimeInstrumentationOptions>? configure = null)
```
Comment (alanwest) [inline src/OpenTelemetry.Instrumentation.Runtime/MeterProviderBuilderExtensions.cs:36]: Related conversation here https://github.com/open-telemetry/opentelemetry-dotnet/pull/3653#discussion_r969915942. I think the direction we'd like to head is to remove optional parameters.

```suggestion
        Action<RuntimeInstrumentationOptions> configure)
```

@CodeBlanch seem correct?

Comment (alanwest) [reply src/OpenTelemetry.Instrumentation.Runtime/MeterProviderBuilderExtensions.cs:36]: @CodeBlanch just reminded me that you'd need to also add a new overload to make this work

```csharp
public static MeterProviderBuilder AddRuntimeInstrumentation(this MeterProviderBuilder builder)
```

Additionally, in the opentelemetry-dotnet repo an named option version of these extensions has been introduced as well. Could be a separate PR.

```csharp
public static MeterProviderBuilder AddRuntimeInstrumentation(
        this MeterProviderBuilder builder,
        string? name,
        Action<RuntimeInstrumentationOptions> configure)
```

Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.Runtime/MeterProviderBuilderExtensions.cs:36]: @alanwest, what is your preference about nullability for cofnigure parameter

```csharp
public static MeterProviderBuilder AddRuntimeInstrumentation(this MeterProviderBuilder builder)
public static MeterProviderBuilder AddRuntimeInstrumentation(this MeterProviderBuilder builder, Action<RuntimeInstrumentationOptions> configure)
```

or

```csharp
public static MeterProviderBuilder AddRuntimeInstrumentation(this MeterProviderBuilder builder)
public static MeterProviderBuilder AddRuntimeInstrumentation(this MeterProviderBuilder builder, Action<RuntimeInstrumentationOptions>? configure)
```

I would keep name attribute as separate PR/issue.

Comment (alanwest) [reply src/OpenTelemetry.Instrumentation.Runtime/MeterProviderBuilderExtensions.cs:36]: The first set is the only one that would work. The second option results in two ambiguous methods.

> I would keep name attribute as separate PR/issue.
👍

Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.Runtime/MeterProviderBuilderExtensions.cs]: Fixed in dff3ace08a4df45a47155a5a57a9fdc3ec90b974.

**Comment by Kielek** [comment]
Fix for failing test potentially in #897

**Review by alanwest — APPROVED** [review]

**Pull request merged**